### PR TITLE
Note with tags and update note tests

### PR DESCRIPTION
**Pull request by nhamhung**
1. Now update Ui with note tags
2. This PR includes EditCommandIntegrationTest and EditCommandParser tests too, to accommodate note tags
3. Refactor some Ui changes: smaller display picture, user detail table always full size and try to hide all scroll-bars (so far in user's dp panel, user's table panel, note list, result display text area and item display, the int, ski, proj, res lists). Not sure if some scroll-bars somewhere else will pop up.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 1]
```diff
@@ -153,19 +153,17 @@ public EditCommand parse(String args) throws ParseException {
...
             }
             if (argMultimap.getValue(PREFIX_TIME).isPresent()) {
-                isAnyFieldEdited = true;
                 editNoteDescriptor.setTime(ParserUtil.parseTime(argMultimap.getValue(PREFIX_TIME).get()));
             }
+            parseTagsForEdit(argMultimap.getAllValues(PREFIX_TAG)).ifPresent(editNoteDescriptor::setTags);
             // ===== End of updating item descriptor =====
 
-            if (!isAnyFieldEdited) {
+            if (!editNoteDescriptor.isAnyFieldEdited()) {
```
Comment (helloImHai) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java:166]: NIce change

File: src/main/resources/view/ResultDisplay.fxml [hunk 2]
```diff
@@ -4,13 +4,16 @@
...
-      </StackPane.margin>
-      <padding>
-         <Insets bottom="10.0" left="10.0" right="10.0" top="10.0" />
-      </padding></TextArea>
+<StackPane fx:id="placeHolder" minHeight="165.0" styleClass="pane-with-border" xmlns="http://javafx.com/javafx/11.0.1" xmlns:fx="http://javafx.com/fxml/1">
+   <VBox>
+      <children>
+          <TextArea fx:id="resultDisplay" editable="false" styleClass="result-display" wrapText="true">
+            <padding>
+               <Insets bottom="5.0" left="5.0" right="5.0" top="5.0" />
```
Comment (helloImHai) [inline src/main/resources/view/ResultDisplay.fxml]: Can rethink these padding numbers? The app looks lopsided and the panel list and the view box does not start together anymore

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 3]
```diff
@@ -1,211 +1,536 @@
...
+                        + PREFIXED_TIME_FROM + PREFIXED_TIME_TO + PREFIXED_INTERNSHIP_DESCRIPTION
+                        + PREFIXED_TAG_FRONTEND,
+                Item.MESSAGE_CONSTRAINTS);
+
+        // invalid item prefix
+        assertParseFailure(parser,
+                VALID_ITEM_INDEX + INVALID_TYPE_DESC + PREFIXED_NAME_GOOGLE + PREFIXED_ROLE_FRONTEND
+                        + PREFIXED_TIME_FROM + PREFIXED_TIME_TO + PREFIXED_INTERNSHIP_DESCRIPTION
+                        + PREFIXED_TAG_FRONTEND,
+                "The item type is invalid!");
```
Comment (chrisjwelly) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: I might have changed the hardcoding of this in another PR. Let me go find later..

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java [hunk 4]
```diff
@@ -1,211 +1,536 @@
...
+        assertParseFailure(parser,
+                VALID_ITEM_INDEX + ITEM_TYPE_PROJECT + PREFIXED_NAME_ORBITAL + PREFIXED_TIME_TO
+                        + INVALID_WEBSITE_DESC + PREFIXED_DESCRIPTION_ORBITAL,
+                Website.MESSAGE_CONSTRAINTS);
+
+        // invalid level
+        assertParseFailure(parser,
+                VALID_ITEM_INDEX + ITEM_TYPE_SKILL + PREFIXED_NAME_REACT
+                        + INVALID_LEVEL_DESC + PREFIXED_TAG_FRONTEND,
+                "A skill level can only be basic, intermediate, or advanced.");
```
Comment (chrisjwelly) [inline src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: I think I also changed this somewhere

**Review by helloImHai — CHANGES_REQUESTED** [review]
As it should be m8 💯💯💯
Good job for the integration part, a suggestion to add tests where the EditXYZDescriptor only has 1 change to it? + Some UI comments

**Review by chrisjwelly — COMMENTED** [review]
I reviewed changes to Parser and ParserTest. I think it looks okay. Same comment as what you told me in my PR #320 , would be good if you could add a test for when you are giving an invalid argument together with a valid argument!

If you have difficulties, it might be because the branch is outdated: please do a `git pull upstream master`?

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java (hunk 3 above)
Comment (chrisjwelly) [reply src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: Right, I made this change in #283 (in Item.java [here](https://github.com/chrisjwelly/main/blob/375547802ff46e85c6a4b3714dde8113103f5b15/src/main/java/seedu/address/model/item/Item.java#L22), to be exact).

So you have the exact same message, but can you change it to `Item.MESSAGE_INVALID_ITEM_TYPE` instead?

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java (hunk 4 above)
Comment (chrisjwelly) [reply src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: I can't remember where I made the change, but you can make use of the class-level constant of `Level` [here](https://github.com/AY1920S2-CS2103T-F10-1/main/blob/887256e2dfb825569b50ada766b38c81166d3bdb/src/main/java/seedu/address/model/item/field/Level.java#L11).

You can use `Level.MESSAGE_CONSTRAINTS` instead

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java (hunk 3 above)
Comment (nhamhung) [reply src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: Thanks I have changed it!

File: src/test/java/seedu/address/logic/parser/EditCommandParserTest.java (hunk 4 above)
Comment (nhamhung) [reply src/test/java/seedu/address/logic/parser/EditCommandParserTest.java]: Yes it's in Level :D

File: src/main/resources/view/ResultDisplay.fxml (hunk 2 above)
Comment (nhamhung) [reply src/main/resources/view/ResultDisplay.fxml]: Yep!!

**Review by chrisjwelly — APPROVED** [review]
Parser seems okay 👍 LGTM!

**Review by helloImHai — APPROVED** [review]
As it should be m8 💯💯💯

**Pull request merged**